spec/lang/machine.md

```diff
@@ -59,6 +60,40 @@ struct CallerReturnInfo<M: Memory> {
 }
 ```
 
+This defines the internal representation of a thread of execution.
+
+```rust
+pub struct Thread<M: Memory> {
+    state: ThreadState,
+
+    // The stack.
```

Comment:
```suggestion
    /// The stack.
```
Same for all the other comments on struct fields etc, please make them doc comments.